src/lib.rs

```diff
@@ -42,6 +42,18 @@ impl FromStr for AnimalName {
         })
     }
 }
+#[cfg(feature = "crypto")]
+impl From<helium_crypto::PublicKey> for AnimalName {
+    fn from(pubkey: helium_crypto::PublicKey) -> Self {
+        let s = pubkey.to_string();
+        let digest = hex_digest(&s);
```

Comment:
call AnimalName::from_str instead of repeating yourself here?

Reply:
I was dancing around the Result that needed unwrapping. I pushed a change